autopush-common/src/db/dual/mod.rs

```diff
@@ -172,14 +183,22 @@ impl DbClient for DualClientImpl {
 
     async fn add_channel(&self, uaid: &Uuid, channel_id: &Uuid) -> DbResult<()> {
         debug!("⚖ getting target");
-        let (target, _) = self.allot(uaid).await?;
+        let (target, is_primary) = self.allot(uaid).await?;
         debug!("⚖ Adding channel to {}", target.name());
-        target.add_channel(uaid, channel_id).await
+        let result = target.add_channel(uaid, channel_id).await;
+        if is_primary {
```

Comment:
These `if is_primary { <write to secondary>` additions as well as some existing others don't include `&& self.write_to_secondary`. Is that intentional?

Reply:
A bit, yes. I figured that we always want to perform those operations. The other ones optionally alter data, but otherwise leave records in tact. Since these are changes to the user record or subscription, I figured we should reflect them.